### PR TITLE
Implement Prime-only features logic

### PR DESCRIPTION
**Pull request by mallardduck**
## Issue: 

 
https://github.com/rancher/rancher/issues/53769

### Implement Prime-only features logic
- Introduce `newPrimeFeature` to mark features as exclusive to Prime builds.
- Convert `RancherSCCRegistrationExtension`, and `ClusterAutoscaling` to Prime-only.
- Force disable Prime features on non-Prime builds in `Enabled` and `IsEnabled`.
- Add unit tests for Prime feature enforcement.

### Update Dashboard logic for Prime-only feature flags
- Stop showing activate button on Prime only features in non-prime build - Future work

### Manual Testing

Tried out the changes in both Prime and Community builds. No obvious issues observed and see expected results. However need to do more testing with edge-case situations like downgrade from Prime to Community.

### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: pkg/features/feature.go [hunk 1]
```diff
@@ -171,16 +171,16 @@ var (
...
 		false,
 		true)
-	OIDCProvider = newFeature(
+	OIDCProvider = newPrimeFeature(
 		"oidc-provider",
 		"Provide an OIDC provider embedded in Rancher. Required to enable SSO in Rancher Prime components.",
-		isPrime(),
+		true,
 		false,
 		true)
```
[the remark this answers is not in the document]
Comment (mallardduck) [reply pkg/features/feature.go]: Fixed this in new push.

**Review by thatmidwesterncoder — DISMISSED** [review]

**Review by joshmeranda — DISMISSED** [review]

File: pkg/features/feature.go [hunk 2]
```diff
@@ -165,16 +165,16 @@ var (
 		true,
 		false,
 		true)
-	OIDCProvider = newFeature(
+	OIDCProvider = newPrimeFeature(
```
Comment (raulcabello) [inline pkg/features/feature.go]: Does this mean community users won't be able to use the `OIDCProvider`? If I recall correctly we wanted to enable by default the `OIDCProvider` for prime users, but we still want to allow community users to use it. Can you confirm @jp-gouin ?

Comment (mallardduck) [reply pkg/features/feature.go]: Oh - good catch @raulcabello. I'm not familiar with OIDC, so will defer to you experts. I can revert this feature being changed once we know for sure.

Comment (mallardduck) [reply pkg/features/feature.go]: Reverted that to regular feature with isPrime for default.

Comment (raulcabello) [reply pkg/features/feature.go]: Thanks! yeah I think it's better to keep the OIDCProvider flag as it was for now. We can always change it later if required

Comment (raulcabello) [reply pkg/features/feature.go]: No blockers from my side since the `OIDCProvider` flag change is reverted, resolving this comment.

Comment (jp-gouin) [reply pkg/features/feature.go]: Nice catch @raulcabello ,yes community users should be able to access it after enabling it. For Prime users it should be on by default

File: pkg/api/norman/customization/feature/feature.go [hunk 3]
```diff
@@ -58,14 +59,21 @@ func Formatter(request *types.APIContext, resource *types.RawResource) {
 }
 
 func getEffectiveValue(resource *types.RawResource) bool {
+	// Prime features always return false on non-Prime builds.
+	status, ok := resource.Values["status"].(map[string]interface{})
+	if ok {
+		if prime, _ := status["prime"].(bool); prime && !features.IsPrime() {
+			return false
+		}
```
Comment (joshmeranda) [inline pkg/api/norman/customization/feature/feature.go:68]: Nit: consider moving the val set into this check. Not a blocker

```suggestion
        }
        
        val, _ = status["default"].(bool)
}
```

Comment (mallardduck) [reply pkg/api/norman/customization/feature/feature.go:68]: Good catch - cleaned that up when I rebased on `main`

**Review by joshmeranda — APPROVED** [review]

**Comment by Negashev** [comment]
(((
